apps/api/src/app/invites/invites.controller.ts

```diff
@@ -37,7 +37,8 @@ export class InvitesController {
     ): Promise<Omit<Invite, "id">> {
         const invite = (await this.invitesService.getInvite(inviteCode)) as any
 
-        invite.group = invite.group.name
+        invite.groupName = invite.group.name
+        invite.groupId = invite.group.id
```

Comment:
Isn't that the groupId of the DB? The on-chain groupId should be different from that groupId, shouldn't it?

Reply:
Yes, this is the groupId in the DB. At the moment, its only used for storing the IDC in local storage for that group.

But, I was thinking of changing the URL to add members from `/group/:groupName/:member` to `/group/:groupId/:member`. So we would need DB group Id here.

Reply:
Umh I think we need to better define our parameters, since having two different group IDs in our codebase can be confusing.

Reply:
THat's one solution. What would happen if we decide to allow admins to update their group names? Should we?

Reply:
I thought of this actually, but its ok either way. We can keep the old ID even if the group name is changed.
ID just have to be a random uint256 so that we can keep the same in DB and in the contract. Hashing name to generate ID instead of going random, was to keep it in sync with how we do it for on-chain groups in Semaphore.

Reply:
Ok let's use the hash of the name then. I was just thinking of 1 scenario:

- Someone changes group name, and keep the old group id = hash(old group name)
- Someone else uses that old group name, but group id = hash(old group name) already exists

We can hash it again in those cases: hash(hash(old group name))

What do you think?

Reply:
Oh, that is a problem. We hash it again I suppose. Or we can simply generate a random one always. Hashing was to keep it similar to on-chain groups but it dont provide any advantage. Anyway hashing is the wrong word here, we were encoding the group name as uin256 for on-chain groups. For off-chain we don't have to decode from ID, so it don't add any benefit 🤔